CMakeLists.txt

```diff
@@ -40,10 +40,8 @@ aux_source_directory(./mozart++ SRCS)
 aux_source_directory(./src SRCS_IMPL)
 
 ## dummy target. DO NOT build this target directly or you will probably get an error.
-add_library(mozart++ STATIC
-        ${SRCS}
-        ${SRCS_IMPL}
-        )
+add_library(mozart++ STATIC ${SRCS_IMPL})
+
```

Comment:
Do not remove `.hpp` files from cmake target.
Let's start from how Makefile builds the project incremently:
1. Collect changed files.
2. Compile changed files.
3. Update timestamp of changed files.

Mozart++ is a template library (or more percisely, a template cross-platform framework), which means
most of our codes are written in header files (namely `.hpp`).
If the cmake target doesn't contain header files, the generated Makefile neither.
So there's the problem we will got: **make will not compile changed headers unless the corresponding source file (namely `.cpp`) are changed**.